validation/gas-token_test.go

```diff
@@ -0,0 +1,105 @@
+package validation
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/stretchr/testify/require"
+)
+
+func testGasToken(t *testing.T, chain *ChainConfig) {
+	skipIfExcluded(t, chain.ChainID)
+
+	client, err := ethclient.Dial(chain.PublicRPC)
+	require.NoError(t, err, "Failed to connect to the Ethereum client at RPC url %s", chain.PublicRPC)
+	defer client.Close()
+
+	// WETH predeploy name() check
+	want := "0000000000000000000000000000000000000000000000000000000000000020" + // offset
+		"000000000000000000000000000000000000000000000000000000000000000d" + // length
+		"5772617070656420457468657200000000000000000000000000000000000000" // "Wrapped Ether" padded to 32 bytes
+	gotName, err := getString("name()", MustHexToAddress("0x4200000000000000000000000000000000000006"), client)
+	require.NoError(t, err)
+	require.Equal(t, want, gotName)
+
+	// L1Block .isCustomGasToken() check
+	got, err := getBool("isCustomGasToken()", MustHexToAddress("0x4200000000000000000000000000000000000015"), client)
+	if !strings.Contains(err.Error(), "execution reverted") {
+		// Pre: Custom Gas Token feature. Reverting is acceptable.
+		require.NoError(t, err)
+	} else {
+		// Post: Custom Gas Token fearure. Must be set to false.
+		require.False(t, got)
+	}
+
+	// SystemConfig .isCustomGasToken() check (L1)
+	client, err = ethclient.Dial(Superchains[chain.Superchain].Config.L1.PublicRPC)
+	require.NoError(t, err, "Failed to connect to the Ethereum client at RPC url %s", chain.PublicRPC)
+	defer client.Close()
+
+	got, err = getBool("isCustomGasToken()", Addresses[chain.ChainID].SystemConfigProxy, client)
+	if !strings.Contains(err.Error(), "execution reverted") {
+		// Pre: Custom Gas Token feature. Reverting is acceptable.
+		require.NoError(t, err)
+	} else {
+		// Post: Custom Gas Token fearure. Must be set to false.
+		require.False(t, got)
+	}
+}
+
+func getString(method string, contractAddress Address, client *ethclient.Client) (string, error) {
```

Comment:
nit, maybe name this `getHexString` or so, because it's not actually a `string` getter (functions that return type `string`) but a bytes getter, which then returns the bytes converted to a hex string.

Reply:
[2ed63de](https://github.com/ethereum-optimism/superchain-registry/pull/399/commits/2ed63de79b4269047f376a7a48f0c54414809dfb)